### PR TITLE
huobi-fetchDepositWithdrawFee(s)

### PR DESCRIPTION
**Pull request by AlgoQ**
# Testing
## Python
`python3 examples/py/cli.py huobi fetchDepositWithdrawFee "USDT"`
```js
Python v3.10.6
CCXT v2.5.44
huobi.fetchDepositWithdrawFee(USDT)
{'deposit': {'fee': None, 'percentage': None},
 'info': {'assetType': '1',
          'chains': [{'addrDepositTag': False,
                      'addrWithTag': False,
                      'chain': 'algousdt',
                      'depositStatus': 'allowed',
                      'displayName': 'ALGO',
                      'fullName': '',
                      'isDynamic': False,
                      'maxWithdrawAmt': '1000000.000000000000000000',
                      'minDepositAmt': '1',
                      'minWithdrawAmt': '2',
                      'numOfConfirmations': '60',
                      'numOfFastConfirmations': '15',
                      'transactFeeWithdraw': '0.001',
                      'withdrawFeeType': 'fixed',
                      'withdrawPrecision': '6',
                      'withdrawQuotaPerDay': '1000000.000000000000000000',
                      'withdrawQuotaPerYear': None,
                      'withdrawQuotaTotal': None,
                      'withdrawStatus': 'allowed'},
                     {'addrDepositTag': False,
                      'addrWithTag': False,
                      'baseChain': 'BSC',
                      'baseChainProtocol': 'bep20',
                      'chain': 'bep20usdt',
                      'depositStatus': 'prohibited',
                      'displayName': 'BEP20',
                      'fullName': '',
                      'isDynamic': False,
                      'maxWithdrawAmt': '1000000.000000000000000000',
                      'minDepositAmt': '1',
                      'minWithdrawAmt': '1',
                      'numOfConfirmations': '30',
                      'numOfFastConfirmations': '15',
                      'transactFeeWithdraw': '0.8',
                      'withdrawFeeType': 'fixed',
                      'withdrawPrecision': '8',
                      'withdrawQuotaPerDay': '1000000.000000000000000000',
                      'withdrawQuotaPerYear': None,
                      'withdrawQuotaTotal': None,
                      'withdrawStatus': 'prohibited'},
                     {'addrDepositTag': False,
                      'addrWithTag': False,
                      'baseChain': 'AVAXCCHAIN',
                      'baseChainProtocol': 'CCHAINERC20',
                      'chain': 'cchainusdt',
                      'depositStatus': 'allowed',
                      'displayName': 'AVAXCCHAIN',
                      'fullName': '',
                      'isDynamic': False,
                      'maxWithdrawAmt': '4000000.000000000000000000',
                      'minDepositAmt': '1',
                      'minWithdrawAmt': '1',
                      'numOfConfirmations': '1',
                      'numOfFastConfirmations': '1',
                      'transactFeeWithdraw': '0.21',
                      'withdrawFeeType': 'fixed',
                      'withdrawPrecision': '6',
                      'withdrawQuotaPerDay': '4000000.000000000000000000',
                      'withdrawQuotaPerYear': None,
                      'withdrawQuotaTotal': None,
                      'withdrawStatus': 'allowed'},
                     {'addrDepositTag': False,
                      'addrWithTag': False,
                      'baseChain': 'CUBE',
                      'baseChainProtocol': 'CUBECRC20',
                      'chain': 'cubeusdt',
                      'depositStatus': 'prohibited',
                      'displayName': 'CUBE',
                      'fullName': '',
                      'isDynamic': True,
                      'maxWithdrawAmt': '100000.000000000000000000',
                      'minDepositAmt': '1',
                      'minWithdrawAmt': '1',
                      'numOfConfirmations': '1',
                      'numOfFastConfirmations': '1',
                      'transactFeeWithdraw': '0.0144834',
                      'withdrawFeeType': 'fixed',
                      'withdrawPrecision': '8',
                      'withdrawQuotaPerDay': '1400000.000000000000000000',
                      'withdrawQuotaPerYear': None,
                      'withdrawQuotaTotal': None,
                      'withdrawStatus': 'prohibited'},
                     {'addrDepositTag': False,
                      'addrWithTag': False,
                      'baseChain': 'HECO',
                      'baseChainProtocol': 'HRC20',
                      'chain': 'hrc20usdt',
                      'depositStatus': 'allowed',
                      'displayName': 'HECO',
                      'fullName': '',
                      'isDynamic': True,
                      'maxWithdrawAmt': '2000000.000000000000000000',
                      'minDepositAmt': '1',
                      'minWithdrawAmt': '2',
                      'numOfConfirmations': '20',
                      'numOfFastConfirmations': '10',
                      'transactFeeWithdraw': '0.04528977',
                      'withdrawFeeType': 'fixed',
                      'withdrawPrecision': '8',
                      'withdrawQuotaPerDay': '2000000.000000000000000000',
                      'withdrawQuotaPerYear': None,
                      'withdrawQuotaTotal': None,
                      'withdrawStatus': 'allowed'},
                     {'addrDepositTag': False,
                      'addrWithTag': False,
                      'baseChain': 'SOLANA',
                      'baseChainProtocol': 'SPL',
                      'chain': 'solusdt',
                      'depositStatus': 'allowed',
                      'displayName': 'SOL',
                      'fullName': '',
                      'isDynamic': False,
                      'maxWithdrawAmt': '1000000.000000000000000000',
                      'minDepositAmt': '1',
                      'minWithdrawAmt': '2',
                      'numOfConfirmations': '300',
                      'numOfFastConfirmations': '100',
                      'transactFeeWithdraw': '1',
                      'withdrawFeeType': 'fixed',
                      'withdrawPrecision': '6',
                      'withdrawQuotaPerDay': '1000000.000000000000000000',
                      'withdrawQuotaPerYear': None,
                      'withdrawQuotaTotal': None,
                      'withdrawStatus': 'allowed'},
                     {'addrDepositTag': False,
                      'addrWithTag': False,
                      'baseChain': 'TRX',
                      'baseChainProtocol': 'TRC20',
                      'chain': 'trc20usdt',
                      'depositStatus': 'allowed',
                      'displayName': 'TRC20',
                      'fullName': '',
                      'isDynamic': False,
                      'maxWithdrawAmt': '1000000.000000000000000000',
                      'minDepositAmt': '1',
                      'minWithdrawAmt': '1',
                      'numOfConfirmations': '1',
                      'numOfFastConfirmations': '1',
                      'transactFeeWithdraw': '0',
                      'withdrawFeeType': 'fixed',
                      'withdrawPrecision': '6',
                      'withdrawQuotaPerDay': '1000000.000000000000000000',
                      'withdrawQuotaPerYear': None,
                      'withdrawQuotaTotal': None,
                      'withdrawStatus': 'allowed'},
                     {'addrDepositTag': False,
                      'addrWithTag': False,
                      'baseChain': 'ETH',
                      'baseChainProtocol': 'ERC20',
                      'chain': 'usdterc20',
                      'depositStatus': 'allowed',
                      'displayName': 'ERC20',
                      'fullName': '',
                      'isDynamic': True,
                      'maxWithdrawAmt': '1000000.000000000000000000',
                      'minDepositAmt': '1',
                      'minWithdrawAmt': '1',
                      'numOfConfirmations': '64',
                      'numOfFastConfirmations': '32',
                      'transactFeeWithdraw': '5.330452',
                      'withdrawFeeType': 'fixed',
                      'withdrawPrecision': '6',
                      'withdrawQuotaPerDay': '1000000.000000000000000000',
                      'withdrawQuotaPerYear': None,
                      'withdrawQuotaTotal': None,
                      'withdrawStatus': 'allowed'}],
          'currency': 'usdt',
          'instStatus': 'normal'},
 'networks': {'ALGO': {'deposit': {'fee': None, 'percentage': None},
                       'withdraw': {'fee': 0.001, 'percentage': False}},
              'AVALANCHE': {'deposit': {'fee': None, 'percentage': None},
                            'withdraw': {'fee': 0.21, 'percentage': False}},
              'BEP20': {'deposit': {'fee': None, 'percentage': None},
                        'withdraw': {'fee': 0.8, 'percentage': False}},
              'CUBE': {'deposit': {'fee': None, 'percentage': None},
                       'withdraw': {'fee': 0.0144834, 'percentage': False}},
              'ERC20': {'deposit': {'fee': None, 'percentage': None},
                        'withdraw': {'fee': 5.330452, 'percentage': False}},
              'HRC20': {'deposit': {'fee': None, 'percentage': None},
                        'withdraw': {'fee': 0.04528977, 'percentage': False}},
              'SOLANA': {'deposit': {'fee': None, 'percentage': None},
                         'withdraw': {'fee': 1.0, 'percentage': False}},
              'TRC20': {'deposit': {'fee': None, 'percentage': None},
                        'withdraw': {'fee': 0.0, 'percentage': False}}},
 'withdraw': {'fee': 0.001, 'percentage': False}}
```

### CODE REVIEW

File: js/huobi.js [hunk 1]
```diff
@@ -7206,6 +7208,123 @@ module.exports = class huobi extends Exchange {
         return this.sortBy (settlements, 'timestamp');
     }
 
+    async fetchDepositWithdrawFees (codes = undefined, params = {}) {
+        /**
+         * @method
+         * @name huobi#fetchDepositWithdrawFees
+         * @description fetch deposit and withdraw fees
+         * @see https://api.hitbtc.com/#currencies
```
Comment (pcriadoperez) [inline js/huobi.js:7216]: Change for huobi

File: js/huobi.js [hunk 2]
```diff
@@ -7206,6 +7208,123 @@ module.exports = class huobi extends Exchange {
...
+        //                  }
+        //              ],
+        //              "instStatus": "normal"
+        //          }
+        //
+        const chains = this.safeValue (fee, 'chains', []);
+        const result = this.depositWithdrawFee (fee);
+        for (let j = 0; j < chains.length; j++) {
+            const chainEntry = chains[j];
+            const networkId = this.safeString (chainEntry, 'displayName');
```
Comment (pcriadoperez) [inline js/huobi.js:7299]: If you look at `fetchCurrencies` we use chain as the networkId
```suggestion
            const networkId = this.safeString (chainEntry, 'chain');
```

File: js/huobi.js [hunk 3]
```diff
@@ -7206,6 +7208,123 @@ module.exports = class huobi extends Exchange {
...
+            const withdrawFeeType = this.safeString (chainEntry, 'withdrawFeeType');
+            const networkCode = this.networkIdToCode (networkId);
```
Comment (pcriadoperez) [inline js/huobi.js:7301]: This should probably be called `this.safeNetwork()` but I would leave that for a different PR.

Comment (carlosmiei) [reply js/huobi.js:7301]: @pcriadoperez actually we had this discussion before and end up preferring this name

File: js/huobi.js [hunk 4]
```diff
@@ -7206,6 +7208,123 @@ module.exports = class huobi extends Exchange {
...
+        //                      "transactFeeWithdraw": "11.1653",
+        //                      "addrWithTag": false,
+        //                      "addrDepositTag": false
+        //                  }
+        //              ],
+        //              "instStatus": "normal"
+        //          }
+        //
+        const chains = this.safeValue (fee, 'chains', []);
+        const result = this.depositWithdrawFee (fee);
```
Comment (carlosmiei) [inline js/huobi.js:7296]: @AlgoQ I think we should assign some default values, for instance, the BTC chain might be the default value for the BTC currency, and so on,

**Comment by carlosmiei** [comment]
LGTM ✅ 

```
 p huobi fetch_deposit_withdraw_fees '["BTC"]'
Python v3.10.8
CCXT v2.6.35
huobi.fetch_deposit_withdraw_fees(['BTC'])
{'BTC': {'deposit': {'fee': None, 'percentage': None},
         'info': {'assetType': '1',
                  'chains': [{'addrDepositTag': False,
                              'addrWithTag': False,
                              'chain': 'btc',
                              'depositStatus': 'allowed',
                              'displayName': 'BTC',
                              'fullName': 'Bitcoin',
                              'isDynamic': False,
                              'maxWithdrawAmt': '100.000000000000000000',
                              'minDepositAmt': '0.0005',
                              'minWithdrawAmt': '0.001',
                              'numOfConfirmations': '2',
                              'numOfFastConfirmations': '1',
                              'transactFeeWithdraw': '0.0004',
                              'withdrawFeeType': 'fixed',
                              'withdrawPrecision': '8',
                              'withdrawQuotaPerDay': '100.000000000000000000',
                              'withdrawQuotaPerYear': None,
                              'withdrawQuotaTotal': None,
                              'withdrawStatus': 'allowed'},
                             {'addrDepositTag': False,
                              'addrWithTag': False,
                              'baseChain': 'ETH',
                              'baseChainProtocol': 'ERC20',
                              'chain': 'hbtc',
                              'depositStatus': 'allowed',
                              'displayName': 'ERC20',
                              'fullName': 'HBTC',
                              'isDynamic': True,
                              'maxWithdrawAmt': '100.000000000000000000',
                              'minDepositAmt': '0.0005',
                              'minWithdrawAmt': '0.001',
                              'numOfConfirmations': '64',
                              'numOfFastConfirmations': '32',
                              'transactFeeWithdraw': '0.00022238',
                              'withdrawFeeType': 'fixed',
                              'withdrawPrecision': '8',
                              'withdrawQuotaPerDay': '100.000000000000000000',
                              'withdrawQuotaPerYear': None,
                              'withdrawQuotaTotal': None,
                              'withdrawStatus': 'allowed'},
                             {'addrDepositTag': False,
                              'addrWithTag': False,
                              'baseChain': 'HECO',
                              'baseChainProtocol': 'HRC20',
                              'chain': 'hrc20btc',
                              'depositStatus': 'allowed',
                              'displayName': 'HECO',
                              'fullName': 'HBTC',
                              'isDynamic': True,
                              'maxWithdrawAmt': '51.228852080000000000',
                              'minDepositAmt': '0.0005',
                              'minWithdrawAmt': '0.001',
                              'numOfConfirmations': '20',
                              'numOfFastConfirmations': '10',
                              'transactFeeWithdraw': '0.00000212',
                              'withdrawFeeType': 'fixed',
                              'withdrawPrecision': '8',
                              'withdrawQuotaPerDay': '100.000000000000000000',
                              'withdrawQuotaPerYear': None,
                              'withdrawQuotaTotal': None,
                              'withdrawStatus': 'prohibited'}],
                  'currency': 'btc',
                  'instStatus': 'normal'},
         'networks': {'BTC': {'deposit': {'fee': None, 'percentage': None},
                              'withdraw': {'fee': 0.0004, 'percentage': False}},
                      'ERC20': {'deposit': {'fee': None, 'percentage': None},
                                'withdraw': {'fee': 0.00022238,
                                             'percentage': False}},
                      'HRC20': {'deposit': {'fee': None, 'percentage': None},
                                'withdraw': {'fee': 2.12e-06,
                                             'percentage': False}}},
         'withdraw': {'fee': 0.0004, 'percentage': False}}}
```

**Review by carlosmiei — APPROVED** [review]
